functional-samples/tutorial.google-analytics/scripts/google-analytics.js

```diff
@@ -0,0 +1,122 @@
+const GA_ENDPOINT = 'https://www.google-analytics.com/mp/collect';
+const GA_DEBUG_ENDPOINT = 'https://www.google-analytics.com/debug/mp/collect';
+
+// Get via https://developers.google.com/analytics/devguides/collection/protocol/ga4/sending-events?client_type=gtag#recommended_parameters_for_reports
+const MEASUREMENT_ID = '<measurement_id>';
+const API_SECRET = '<api_secret>';
+const DEFAULT_ENGAGEMENT_TIME_MSEC = 100;
+
+// Duration of inactivity after which a new session is created
+const SESSION_EXPIRATION_IN_MIN = 30;
+
+export class Analytics {
+  constructor(debug = false) {
+    this.debug = debug;
+  }
+
+  // Returns the client ID, or creates a new one if one doesn't exist.
+  // Stores client id in local storage to keep the same client_id as long as
+  // the extension is installed.
+  async getOrCreateClientId() {
+    const result = await chrome.storage.local.get('clientId');
+    let clientId = result.clientId;
+    if (!clientId) {
+      // Generate a unique client ID, the actual value is not relevant
+      clientId = self.crypto.randomUUID();
+      await chrome.storage.local.set({ clientId });
+    }
+    return clientId;
+  }
+
+  // Returns the current session ID, or creates a new one if one doesn't exist or
+  // the previous one has expired.
+  async getOrCreateSessionId() {
+    // Store session in memory storage
+    const result = await chrome.storage.session.get('sessionData');
```

Comment:
since you never use result again, `const {sessionData} = await chrome.storage.session.get('sessionData');`